File: csharp/Models/Enums.cs
using System.Text.Json.Serialization;

namespace NewNanManager.Client.Models;

/// <summary>
/// 封禁模式枚举
/// </summary>
public enum BanMode
{
    /// <summary>
    /// 正常状态（未封禁）
    /// </summary>
    Normal = 0,

    /// <summary>
    /// 临时封禁
    /// </summary>
    Temporary = 1,

    /// <summary>
    /// 永久封禁
    /// </summary>
    Permanent = 2,
}

/// <summary>
/// 登录动作枚举
/// </summary>
[JsonConverter(typeof(JsonStringEnumConverter))]
public enum LoginAction
{
    /// <summary>
    /// 登录
    /// </summary>
    Login,

    /// <summary>
    /// 登出
    /// </summary>
    Logout,
}

/// <summary>
/// 服务器类型枚举
/// </summary>
[JsonConverter(typeof(JsonStringEnumConverter))]
public enum ServerType
{
    /// <summary>
    /// Minecraft游戏服务器
    /// </summary>
    Minecraft,

    /// <summary>
    /// 代理服务器
    /// </summary>
    Proxy,

    /// <summary>
    /// 大厅服务器
    /// </summary>
    Lobby,
}

/// <summary>
/// IP威胁等级枚举
/// </summary>
public enum ThreatLevel
{
    /// <summary>
    /// 低威胁
    /// </summary>
    Low = 0,

    /// <summary>
    /// 中等威胁
    /// </summary>
    Medium = 1,

    /// <summary>
    /// 高威胁
    /// </summary>
    High = 2,

    /// <summary>
    /// 严重威胁
    /// </summary>
    Critical = 3,
}

/// <summary>
/// IP查询状态枚举
/// </summary>
public enum QueryStatus
{
    /// <summary>
    /// 待查询
    /// </summary>
    Pending = 0,

    /// <summary>
    /// 已完成
    /// </summary>
    Completed = 1,

    /// <summary>
    /// 查询失败
    /// </summary>
    Failed = 2,
}

File: python/newnanmanager/models/enums.py
"""Enumeration types for NewNanManager API."""

from enum import IntEnum


class BanMode(IntEnum):
    """封禁模式枚举."""

    NORMAL = 0  # 正常状态（未封禁）
    TEMPORARY = 1  # 临时封禁
    PERMANENT = 2  # 永久封禁


class LoginAction(IntEnum):
    """登录动作枚举."""

    LOGIN = 1  # 登录
    LOGOUT = 2  # 登出


class ThreatLevel(IntEnum):
    """IP威胁等级枚举."""

    LOW = 0  # 低威胁
    MEDIUM = 1  # 中等威胁
    HIGH = 2  # 高威胁
    CRITICAL = 3  # 严重威胁


class QueryStatus(IntEnum):
    """IP查询状态枚举."""

    PENDING = 0  # 待查询
    COMPLETED = 1  # 已完成
    FAILED = 2  # 查询失败
